src/domain/pricing/bulk-editor/index.tsx

```diff
@@ -0,0 +1,1052 @@
+      if (
+        confirm(
+          "There are unsaved changes. Do you want to discard and continue?"
+        )
+      ) {
+        setPriceChanges({})
+        return true
+      } else {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  const onPriceListSelect = (priceListId: string) => {
+    if (checkForChanges()) {
+      setCurrentPriceListId(priceListId)
+    }
+  }
+
+  const onClose = () => {
+    if (checkForChanges()) {
+      closeEditor()
+    }
+  }
+
+  /**
+   * Save changes callback.
+   * Iterate over a price changes object and prepare MA prices for bulk update.
+   */
+  const onSave = () => {
+    const prices: Partial<MoneyAmount>[] = []
+
+    const _priceChanges = { ...priceChanges }
+
+    priceList.prices
+      .map((p) =>
+        pick(p, [
+          "amount",
+          "variant_id",
+          "region_id",
+          "min_quantity",
+          "max_quantity",
+          "currency_code",
+        ])
+      )
+      .map((plPrice) => {
+        if (
+          plPrice.region_id &&
+          plPrice.min_quantity === null &&
+          plPrice.max_quantity === null
+        ) {
+          const priceKey = getPriceKey(plPrice.variant_id, plPrice.region_id)
+          if (priceKey in _priceChanges) {
+            // MA is deleted
+            if (_priceChanges[priceKey] === undefined) {
+              return
+            }
+
+            // MA is updated
+
+            let preparedAmount = Math.round(
+              parseFloat(_priceChanges[priceKey]) *
+                10 **
+                  currencies[plPrice.currency_code.toUpperCase()].decimal_digits
+            )
+
+            if (isNaN(preparedAmount)) {
+              return
+            }
+
+            prices.push({
+              variant_id: plPrice.variant_id,
+              region_id: plPrice.region_id,
+              amount: preparedAmount,
+            })
+
+            delete _priceChanges[priceKey]
+          } else {
+            prices.push(plPrice)
+          }
+        } else {
+          prices.push(plPrice)
+        }
+      })
+
+    // entries that are left --> MAs to be created
+    Object.entries(_priceChanges).map(([k, amount]) => {
+      const [variantId, regionId] = k.split("-")
+
+      const reg = regions?.find((r) => r.id === regionId)!
+      const curr = currencies[reg.currency_code.toUpperCase()]
+
+      const preparedAmount = Math.round(
+        parseFloat(amount) *
+          10 ** currencies[curr.code.toUpperCase()].decimal_digits
+      )
+
+      if (isNaN(preparedAmount)) {
+        return
+      }
+
+      prices.push({
+        variant_id: variantId,
+        region_id: regionId,
+        amount: preparedAmount,
+      })
+    })
+
+    prices.forEach((p) => {
+      delete p.id
+      if (p.region_id) {
+        delete p.currency_code
+      }
+    }) // new records will be created
+
+    updatePriceList.mutate(
+      { prices, override: true },
+      { onSuccess: () => closeEditor() }
```

Comment:
Are we sure that we want to close the editor automatically after a save? should we just show a message and let the user decide. I am thinking the case the user just want to save middleway and then continue few minutes later for example (after a break :)). He would have to re open the editor and config the fields to show.

Reply:
Hmm a question for Ludvig for sure but since this form behaves as a modal - it is usually expected that modals dismiss if an action is successfully completed.

Reply:
Yeah I agree, but generally, the modal hold a form that you have to fully complete before saving. Here we are in a case where you can do it in multiple time. It is very much a question ^*